src/adapter/src/catalog.rs

```diff
@@ -7785,6 +7868,49 @@ impl Catalog {
         }
     }
 
+    pub(crate) fn to_serialized_item(item: CatalogItem) -> SerializedCatalogItem {
```

Comment:
Almost identical to the method directly above.

Reply:
Can you get rid of the one above and any call sites that pass a reference, call `.clone()` on their args?

Reply:
Joe explained the problem to me offline, we want to avoid cloning a `CatalogItem` because they're quite large, but if we already have an owned `CatalogItem` we want to re-use the string.

Second proposal create methods on `CatalogItem`, and use the `to_*` and `into_*` prefixes which are part of the [Rust API naming guidelines](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv).

```
impl CatalogItem {
  fn to_serialized(&self) -> SerializedCatalogItem {
     
  }
  fn into_serialized(self) -> SerializedCatalogItem {

  }
}
``